src/main/java/org/datacleaner/kettle/configuration/DataCleanerConfigurationDialog.java

```diff
@@ -150,41 +146,36 @@ public void widgetSelected(SelectionEvent event) {
                 if (dir != null) {
-                    _text.setText(dir);
-                    try {
-                        final SoftwareVersion editionDetails = SoftwareVersionHelper.getEditionDetails(dir);
-                        if (editionDetails != null) {
-                            labelEdition.setText(EDITION.trim() + " " + editionDetails.getName());
-                            labelVersion.setText(VERSION.trim() + " " + editionDetails.getVersion());
-                            errorLabel.setVisible(false);
-                            _result = dir;
-                        } else {
-                            errorLabel.setVisible(true);
-                            _result = null;
-                        }
-                    } catch (IOException e) {
-                        errorLabel.setText("Exception while reading the directory");
-                    }
-
+                    setDataCleanerInstallationFolder(dir);
                 }
             }
         });
 
         try {
-            final String pluginFolderPath = ModelerHelper.getPluginFolderPath();
-            final String dcInstallationFolder = ModelerHelper.getDataCleanerInstalationPath(pluginFolderPath);
-            if (dcInstallationFolder != null) {
-                _text.setText(dcInstallationFolder);
-            }
-            final SoftwareVersion editionDetails = SoftwareVersionHelper.getEditionDetails(dcInstallationFolder);
+            DataCleanerSpoonConfiguration dcSpoonConfiguration = DataCleanerSpoonConfiguration.load();
+            setDataCleanerInstallationFolder(dcSpoonConfiguration.getDataCleanerInstallationFolderPath());
+        } catch (DataCleanerSpoonConfigurationException e) {
+            // Do nothing if a previous configuration could not be loaded
+        }
+    }
+
+    protected void setDataCleanerInstallationFolder(String dir) {
+        _text.setText(dir);
+        try {
+            final DataCleanerSpoonConfiguration dataCleanerSpoonConfiguration = new DataCleanerSpoonConfiguration(null,
+                    dir);
+            final SoftwareVersion editionDetails = SoftwareVersionHelper
+                    .getEditionDetails(dataCleanerSpoonConfiguration);
             if (editionDetails != null) {
-                labelEdition.setText(EDITION.trim() + " " + editionDetails.getName());
-                labelVersion.setText(VERSION.trim() + " " + editionDetails.getVersion());
+                _labelEdition.setText(EDITION.trim() + " " + editionDetails.getName());
+                _labelVersion.setText(VERSION.trim() + " " + editionDetails.getVersion());
+                _errorLabel.setVisible(false);
+            } else {
+                _errorLabel.setVisible(true);
             }
-        } catch (Throwable e) {
-            // Do nothing If the file doesn't exit 
+        } catch (IOException e) {
+            _errorLabel.setText("Exception while reading the directory");
         }
```

Comment:
If there is an IO Exception I believe we shouldn't show anything to the user.  You could log it if you want. The original story is this one: https://github.com/datacleaner/pdi-datacleaner/issues/10. Basically you wanted to show the settings. However, if the user has never configured the DC options before, the file doesn't exist and a exception will be thrown.  That's the reason I decided not to show any message. Also I did not considered that it is necessary to log it.